src/music-player/core/util/musiclyric.cpp

```diff
@@ -90,13 +91,36 @@ static QString getFileCodex(QString dir)
     return code;
 }
 
+void MusicLyric::parseLyric(const QString &str)
+{
+    auto lines = str.split("\n");
+    QRegExp rx("\\[([^\\]]*)\\]\\s*(\\S.*)");
+    QVector<QPair<qint64,QString>> tmp;
+    for(auto line : lines) {
+        rx.indexIn(line);
+        auto timeStr = rx.capturedTexts()[1];
+        auto lyricStr = rx.capturedTexts()[2];
+        QTime t = QTime::fromString(timeStr,"mm:ss.z");
+        qint64 time = t.msecsSinceStartOfDay();
+        if(t.isValid())
+            tmp.push_back({time,lyricStr});
+    }
+    std::sort(tmp.begin(),tmp.end());
+    this->position.clear();
+    this->line.clear();
+    for(auto item:tmp) {
+        this->position.push_back(item.first);
+        this->line.push_back(item.second);
```

Comment:
`this->line.push_back(item.second.trimmed());`

see https://github.com/linuxdeepin/deepin-music/pull/112

Reply:
or we can fix the RegExp to `\\[([^\\]]*)\\]\\s*(\\S.*\\S)\\s*$`
which works as
![image](https://user-images.githubusercontent.com/38401412/150278858-4b892a1a-e6ba-4c00-a639-e262f3d2ce5a.png)

Reply:
cool